server/src/main/java/org/elasticsearch/common/lucene/MinimumScoreCollector.java

```diff
@@ -15,9 +15,15 @@
 import org.apache.lucene.search.ScoreCachingWrappingScorer;
 import org.apache.lucene.search.ScoreMode;
 import org.apache.lucene.search.SimpleCollector;
+import org.apache.lucene.search.Weight;
 
 import java.io.IOException;
 
+/**
+ * Collector that wraps another collector and collects only documents that have a score that's greater or equal than the
+ * provided minimum score. Given that this collector filters documents out, it does and should not override {@link #setWeight(Weight)},
+ * as that may lead to exposing total hit count that does not reflect the filtering.
```

Comment:
Thanks for adding this doc. I found it helpful :). Can we override and make the `setWeight` method final this class and have this comment in the method instead? It's totally optional.

Reply:
I will take these two out of this PR and open another one.